irohad/ametsuchi/reconnection/reconnection_strategy.hpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_RECONNECTION_STRATEGY_HPP
+#define IROHA_RECONNECTION_STRATEGY_HPP
+
+#include <string>
+
+namespace iroha {
+  namespace ametsuchi {
+    /**
+     * The interface provides a strategy for reconnection to data base.
+     */
+    class ReconnectionStorageStrategy {
+     public:
+      /// Identifier type of unique invocation
+      using Tag = std::string;
```

Comment:
I think we could use something more lighweight than strings. some POD. Is there a reason to mess up with strings? also, current `makeTag` may have collisions if prefix ends with a digit - an unneeded reason to worry because of `string` tags...

Reply:
to be more constructive, it could be a
```c++
union {
  uint64_t all;
  struct {
    uint32_t requestor;
    uint32_t counter;
  };
};
```
of course, the union is needed if we need the `requestor` logic that is realized by the string prefix now. if we could do without it things would be even simpler - `Tag` can become a `size_t`.